ExtensionConfigurationException on system properties annotations example provided in the doc

I copied the [example](https://junit-pioneer.org/docs/system-properties/) provided in the doc to a [test class in my project](https://github.com/oliviercailloux/JARiS/blob/pioneer/src/test/java/io/github/oliviercailloux/jaris/credentials/MySystemPropertyTest.java). When executing `mvn test`, I get this [exception from pioneer](https://github.com/oliviercailloux/JARiS/runs/2487957466?check_suite_focus=true#step:5:40).

````
[INFO] Running io.github.oliviercailloux.jaris.credentials.MySystemPropertyTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.006 s <<< FAILURE! - in io.github.oliviercailloux.jaris.credentials.MySystemPropertyTest
[ERROR] test  Time elapsed: 0 s  <<< ERROR!
org.junit.jupiter.api.extension.ExtensionConfigurationException: Don't clear/set the same entry more than once.
Caused by: java.lang.IllegalStateException: Cannot clear and set the following entries at the same time: some property
````

Here are my dependencies.
````
[INFO] --- maven-dependency-plugin:3.1.2:tree (default-cli) @ jaris ---
[INFO] io.github.oliviercailloux:jaris:jar:0.0.10-SNAPSHOT
[INFO] +- org.junit-pioneer:junit-pioneer:jar:1.4.0:test
[INFO] |  +- org.junit.jupiter:junit-jupiter-api:jar:5.7.1:test
[INFO] |  |  \- org.opentest4j:opentest4j:jar:1.2.0:test
[INFO] |  +- org.junit.jupiter:junit-jupiter-params:jar:5.7.1:test
[INFO] |  +- org.junit.platform:junit-platform-commons:jar:1.7.1:test
[INFO] |  \- org.junit.platform:junit-platform-launcher:jar:1.7.1:test
[INFO] +- com.google.jimfs:jimfs:jar:1.2:test
[INFO] +- org.mockito:mockito-core:jar:3.9.0:test
[INFO] |  +- net.bytebuddy:byte-buddy:jar:1.10.20:test
[INFO] |  +- net.bytebuddy:byte-buddy-agent:jar:1.10.20:test
[INFO] |  \- org.objenesis:objenesis:jar:3.2:test
[INFO] +- org.slf4j:slf4j-api:jar:1.7.30:compile
[INFO] +- com.google.guava:guava:jar:30.1.1-jre:compile
[INFO] |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO] |  +- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO] |  +- org.checkerframework:checker-qual:jar:3.8.0:compile
[INFO] |  +- com.google.errorprone:error_prone_annotations:jar:2.5.1:compile
[INFO] |  \- com.google.j2objc:j2objc-annotations:jar:1.3:compile
[INFO] +- ch.qos.logback:logback-classic:jar:1.2.3:test
[INFO] |  \- ch.qos.logback:logback-core:jar:1.2.3:test
[INFO] \- org.junit.jupiter:junit-jupiter-engine:jar:5.7.1:test
[INFO]    +- org.apiguardian:apiguardian-api:jar:1.1.0:test
[INFO]    \- org.junit.platform:junit-platform-engine:jar:1.7.1:test
````

I must be missing something stupid, but I can’t find out what.

--- a/README.md
+++ b/README.md
@@ -111,7 +111,7 @@ The least we can do is to thank them and list some of their accomplishments here
 #### 2021
 
 * [Cory Thomas](https://github.com/dump247) contributed the `minSuccess` attribute in [retrying tests](https://junit-pioneer.org/docs/retrying-test/) (#408 / #430)
-* [Daniel Kraus](https://github.com/beatngu13) fixed bugs in the environment variable and system property extensions (#432 / #433, #448 / #449 and #473 / #476)
+* [Daniel Kraus](https://github.com/beatngu13) fixed bugs in the environment variable and system property extensions (#432 / #433, #448 / #449, and #473 / #476)
 * [Slawomir Jaranowski](https://github.com/slawekjaranowski) Migrate to new Shipkit plugins (#410 / #419)
 * [Stefano Cordio](https://github.com/scordio) contributed [the Cartesian Enum source](https://junit-pioneer.org/docs/cartesian-product/#cartesianenumsource) (#379 / #409 and #414 / #453)
 

--- a/README.md
+++ b/README.md
@@ -111,7 +111,7 @@ The least we can do is to thank them and list some of their accomplishments here
 #### 2021
 
 * [Cory Thomas](https://github.com/dump247) contributed the `minSuccess` attribute in [retrying tests](https://junit-pioneer.org/docs/retrying-test/) (#408 / #430)
-* [Daniel Kraus](https://github.com/beatngu13) fixed bugs in the environment variable and system property extensions (#432 / #433 and #448 / #449)
+* [Daniel Kraus](https://github.com/beatngu13) fixed bugs in the environment variable and system property extensions (#432 / #433, #448 / #449 and #473 / #476)
 * [Slawomir Jaranowski](https://github.com/slawekjaranowski) Migrate to new Shipkit plugins (#410 / #419)
 * [Stefano Cordio](https://github.com/scordio) contributed [the Cartesian Enum source](https://junit-pioneer.org/docs/cartesian-product/#cartesianenumsource) (#379 / #409 and #414 / #453)
 

--- a/src/main/java/org/junitpioneer/jupiter/EnvironmentVariableExtension.java
+++ b/src/main/java/org/junitpioneer/jupiter/EnvironmentVariableExtension.java
@@ -17,6 +17,7 @@ import java.util.Set;
 import java.util.concurrent.atomic.AtomicBoolean;
 
 import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.platform.commons.support.AnnotationSupport;
 import org.junitpioneer.internal.PioneerAnnotationUtils;
 import org.junitpioneer.internal.PioneerUtils;
 
@@ -36,16 +37,18 @@ class EnvironmentVariableExtension extends AbstractEntryBasedExtension<String, S
 
 	@Override
 	protected Set<String> entriesToClear(ExtensionContext context) {
-		return PioneerAnnotationUtils
-				.findClosestEnclosingRepeatableAnnotations(context, ClearEnvironmentVariable.class)
+		return AnnotationSupport
+				.findRepeatableAnnotations(context.getElement(), ClearEnvironmentVariable.class)
+				.stream()
 				.map(ClearEnvironmentVariable::key)
 				.collect(PioneerUtils.distinctToSet());
 	}
 
 	@Override
 	protected Map<String, String> entriesToSet(ExtensionContext context) {
-		return PioneerAnnotationUtils
-				.findClosestEnclosingRepeatableAnnotations(context, SetEnvironmentVariable.class)
+		return AnnotationSupport
+				.findRepeatableAnnotations(context.getElement(), SetEnvironmentVariable.class)
+				.stream()
 				.collect(toMap(SetEnvironmentVariable::key, SetEnvironmentVariable::value));
 	}
 

--- a/src/main/java/org/junitpioneer/jupiter/EnvironmentVariableExtension.java
+++ b/src/main/java/org/junitpioneer/jupiter/EnvironmentVariableExtension.java
@@ -38,6 +38,10 @@ class EnvironmentVariableExtension extends AbstractEntryBasedExtension<String, S
 	@Override
 	protected Set<String> entriesToClear(ExtensionContext context) {
 		return AnnotationSupport
+				// This extension implements `BeforeAllCallback` and `BeforeEachCallback` and so if an outer class
+				// (i.e. a class that the test class is @Nested within) uses this extension, this method will be
+				// called on those extension points and discover the variables to set/clear. That means we don't need
+				// to search for enclosing annotations here.
 				.findRepeatableAnnotations(context.getElement(), ClearEnvironmentVariable.class)
 				.stream()
 				.map(ClearEnvironmentVariable::key)
@@ -47,6 +51,10 @@ class EnvironmentVariableExtension extends AbstractEntryBasedExtension<String, S
 	@Override
 	protected Map<String, String> entriesToSet(ExtensionContext context) {
 		return AnnotationSupport
+				// This extension implements `BeforeAllCallback` and `BeforeEachCallback` and so if an outer class
+				// (i.e. a class that the test class is @Nested within) uses this extension, this method will be
+				// called on those extension points and discover the variables to set/clear. That means we don't need
+				// to search for enclosing annotations here.
 				.findRepeatableAnnotations(context.getElement(), SetEnvironmentVariable.class)
 				.stream()
 				.collect(toMap(SetEnvironmentVariable::key, SetEnvironmentVariable::value));

--- a/src/main/java/org/junitpioneer/jupiter/SystemPropertyExtension.java
+++ b/src/main/java/org/junitpioneer/jupiter/SystemPropertyExtension.java
@@ -16,6 +16,7 @@ import java.util.Map;
 import java.util.Set;
 
 import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.platform.commons.support.AnnotationSupport;
 import org.junitpioneer.internal.PioneerAnnotationUtils;
 import org.junitpioneer.internal.PioneerUtils;
 
@@ -29,16 +30,18 @@ class SystemPropertyExtension extends AbstractEntryBasedExtension<String, String
 
 	@Override
 	protected Set<String> entriesToClear(ExtensionContext context) {
-		return PioneerAnnotationUtils
-				.findClosestEnclosingRepeatableAnnotations(context, ClearSystemProperty.class)
+		return AnnotationSupport
+				.findRepeatableAnnotations(context.getElement(), ClearSystemProperty.class)
+				.stream()
 				.map(ClearSystemProperty::key)
 				.collect(PioneerUtils.distinctToSet());
 	}
 
 	@Override
 	protected Map<String, String> entriesToSet(ExtensionContext context) {
-		return PioneerAnnotationUtils
-				.findClosestEnclosingRepeatableAnnotations(context, SetSystemProperty.class)
+		return AnnotationSupport
+				.findRepeatableAnnotations(context.getElement(), SetSystemProperty.class)
+				.stream()
 				.collect(toMap(SetSystemProperty::key, SetSystemProperty::value));
 	}
 

--- a/src/main/java/org/junitpioneer/jupiter/SystemPropertyExtension.java
+++ b/src/main/java/org/junitpioneer/jupiter/SystemPropertyExtension.java
@@ -31,6 +31,10 @@ class SystemPropertyExtension extends AbstractEntryBasedExtension<String, String
 	@Override
 	protected Set<String> entriesToClear(ExtensionContext context) {
 		return AnnotationSupport
+				// This extension implements `BeforeAllCallback` and `BeforeEachCallback` and so if an outer class
+				// (i.e. a class that the test class is @Nested within) uses this extension, this method will be
+				// called on those extension points and discover the properties to set/clear. That means we don't need
+				// to search for enclosing annotations here.
 				.findRepeatableAnnotations(context.getElement(), ClearSystemProperty.class)
 				.stream()
 				.map(ClearSystemProperty::key)
@@ -40,6 +44,10 @@ class SystemPropertyExtension extends AbstractEntryBasedExtension<String, String
 	@Override
 	protected Map<String, String> entriesToSet(ExtensionContext context) {
 		return AnnotationSupport
+				// This extension implements `BeforeAllCallback` and `BeforeEachCallback` and so if an outer class
+				// (i.e. a class that the test class is @Nested within) uses this extension, this method will be
+				// called on those extension points and discover the properties to set/clear. That means we don't need
+				// to search for enclosing annotations here.
 				.findRepeatableAnnotations(context.getElement(), SetSystemProperty.class)
 				.stream()
 				.collect(toMap(SetSystemProperty::key, SetSystemProperty::value));

--- a/src/test/java/org/junitpioneer/jupiter/EnvironmentVariableExtensionTests.java
+++ b/src/test/java/org/junitpioneer/jupiter/EnvironmentVariableExtensionTests.java
@@ -158,6 +158,19 @@ class EnvironmentVariableExtensionTests {
 			assertThat(systemEnvironmentVariable("clear envvar F")).isNull();
 		}
 
+		@Test
+		@DisplayName("method level should not clash (in terms of duplicate entries) with class level")
+		@SetEnvironmentVariable(key = "set envvar A", value = "new A")
+		void methodLevelShouldNotClashWithClassLevel() {
+			assertThat(systemEnvironmentVariable("set envvar A")).isEqualTo("new A");
+			assertThat(systemEnvironmentVariable("set envvar B")).isEqualTo("old B");
+			assertThat(systemEnvironmentVariable("set envvar C")).isEqualTo("old C");
+			assertThat(systemEnvironmentVariable("clear envvar D")).isEqualTo("new D");
+
+			assertThat(systemEnvironmentVariable("clear envvar E")).isNull();
+			assertThat(systemEnvironmentVariable("clear envvar F")).isNull();
+		}
+
 	}
 
 	@DisplayName("with nested classes")

--- a/src/test/java/org/junitpioneer/jupiter/SystemPropertyExtensionTests.java
+++ b/src/test/java/org/junitpioneer/jupiter/SystemPropertyExtensionTests.java
@@ -150,6 +150,19 @@ class SystemPropertyExtensionTests {
 			assertThat(System.getProperty("clear prop F")).isNull();
 		}
 
+		@Test
+		@DisplayName("method level should not clash (in terms of duplicate entries) with class level")
+		@SetSystemProperty(key = "set prop A", value = "new A")
+		void methodLevelShouldNotClashWithClassLevel() {
+			assertThat(System.getProperty("set prop A")).isEqualTo("new A");
+			assertThat(System.getProperty("set prop B")).isEqualTo("old B");
+			assertThat(System.getProperty("set prop C")).isEqualTo("old C");
+			assertThat(System.getProperty("clear prop D")).isEqualTo("new D");
+
+			assertThat(System.getProperty("clear prop E")).isNull();
+			assertThat(System.getProperty("clear prop F")).isNull();
+		}
+
 	}
 
 	@DisplayName("with nested classes")

